src/main/java/net/sf/jabref/model/groups/SearchGroup.java

```diff
@@ -21,18 +21,26 @@
 
     public static final String ID = "SearchGroup:";
 
-    private final SearchQuery query;
+    private final GroupSearchQuery query;
 
     private static final Log LOGGER = LogFactory.getLog(SearchGroup.class);
 
+    private final String searchExpression;
+    private final boolean caseSensitive;
+    private final boolean regExp;
+
+
     /**
      * Creates a SearchGroup with the specified properties.
      */
     public SearchGroup(String name, String searchExpression, boolean caseSensitive, boolean regExp,
             GroupHierarchyType context) {
         super(name, context);
 
-        this.query = new SearchQuery(searchExpression, caseSensitive, regExp);
+        this.searchExpression = searchExpression;
```

Comment:
I think it is enough that these information are stored in the query. What was your idea behind these changes?